docs/digger/developer/jenkinsfile.asciidoc

```diff
@@ -0,0 +1,166 @@
+    }
+  }
+  ...
+}
+----
+
+The `signAndroidApks` function takes the following parameters.
+
+.signAndroidApks options
+|===
+| Parameter | Description
+
+| keyStoreId
+| Keystore ID. This will be provided through the `BUILD_CREDENTIAL_ID` parameter.
+
+| keyAlias
+| Alias of the private key/certificate chain. This will be provided through the `BUILD_CREDENTIAL_ALIAS` parameter.
+
+| apksToSign
+| A file name or glob pattern specifying the APK files to store.
+
+| signedApkMapping
+| Output the signed APK to a separate directory than the unsigned APK. Omit if storing in the same directory.
+
+| androidHome
+| Override Android home directory.
+|===
+
+=== Building projects with Xcode
+To build Xcode projects you can use the `xcodeBuild` function in your
+Jenkinsfile, this requires the `XC_VERSION` environment variable to be defined.
+To sign the app the `xcodeBuild` function can then be used.
+
+[source,groovy]
+----
+node('ios'){
+  ...
+  stage('Build') {
+    withEnv(['XC_VERSION=8.3']) {
+      xcodeBuild(
+        cleanBeforeBuild: true,
+        src: 'helloworld-ios-swift',
+        schema: 'helloworld-ios-app',
+        workspace: 'helloworld-ios-app',
+        buildDir: 'build',
+        sdk: 'iphoneos',
+        version: '0.1-alpha',
+        shortVersion: '0.1',
+        bundleId: 'com.feedhenry.helloworld-ios-app',
+        infoPlistPath: 'helloworld-ios-app/helloworld-ios-app-Info.plist',
+        flags: '-fstack-protector -fstack-protector-all ENABLE_BITCODE=NO',
+        autoSign: false
+      )
+    }
+  }
+
+  stage('CodeSign') {
+    codeSign(
+      profileId: 'redhat-dist-dp',
+      clean: true,
+      verify: true,
+      ipaName: 'myapp',
+      appPath: 'helloworld-ios-swift/build/Debug-iphoneos/helloworld-ios-app.app'
+    )
+  }
+  ...
+}
+----
+
+The funtion takes the following parameters.
+
+.xcodeBuild options
+|===
+| Parameter | Description
+
+| cleanBeforeBuild
+|	This will delete the build directories before invoking the build. This will force the rebuilding of all dependencies and can make large projects take a lot longer
+
+| target
+| The target to build. If left empty, this will build all targets in the project. If you wish to build your binary and the unit test module, it is best to do this as two separate steps each with their own target.
+
+| sdk
+| You only need to supply this value if you want to specify the SDK to build against. If empty, the SDK will be determined by XCode.
+
+| workspace
+| Workspace to build from.
+
+| buildDir
+| The value to use for CONFIGURATION_BUILD_DIR setting (BUILD_DIR in >= 2.0.0).
+
+| version
+| Version of the app.
+
+| shortVersion
+| Short representation of the apps version.
+
+| autoSign
+| Whether to sign the app automatically or not.
+
+| infoPlistPath
+| Path to the apps plist file.
+
+| flags
+| Flags to add to the command that is executed.
+|===
+
+.codeSign options
+|===
+| Parameter | Description
+
+| profileId
+| ID of the developer profile.
```

Comment:
I think we could mention that our digger jenkins client can send this value as a build parameter